### PR TITLE
Internationalization with react-i18next

### PR DESCRIPTION
**Pull request by kwingram25**
- Use react-i18next and translated strings throughout the app
- Fix pre-commit hook(s)

### CODE REVIEW

**Comment by achimcc** [comment]
@kwingram25 : If we add i18next, can we also add a switch to select the locale/language in the Settings page of the App in the same PR? If you add a de/german locale as a second language, I'm happy to provide the translation. You can just copy/paste the english text and I'll replace it later on or send you the modified json file. Otherwise, I'm afraid we can't test the internationalization properly?

**Comment by achimcc** [comment]
I tested German translation and it looks like we are currently translating only a part of the App?

image

E.g. the headline saying 'Upload and instantiate Contract' doesn't get translated. As same as the Stepper on the right side of the instantiation wizard.

**Comment by kwingram25** [comment]
That just means I forgot to add the t hook in certain places.

**Comment by achimcc** [comment]
I added translations for the newly created fields in the JSON file. But there is still a lot of untranslated text in the App. Also including the translations I mentioned above.

If we add translations, in my opinion we should make sure that we translate all texts in the App.

**Comment by achimcc** [comment]
I went through the code and added the missing translations.

File: src/ui/hooks/useLocalStorage.ts [hunk 1]
```diff
@@ -8,7 +8,11 @@ export function useLocalStorage<T>(key: string, initialValue: T): [T, (_: T) =>
     try {
       const item = window.localStorage.getItem(key);
 
-      return item ? (JSON.parse(item) as T) : initialValue;
+      return item
```
Comment (achimcc) [inline src/ui/hooks/useLocalStorage.ts:11]: Also the things happening here, it is not really clear why these changes are happening and I think they are not part of translating the app?

File: src/ui/hooks/useNonZeroBn.ts [hunk 2]
```diff
@@ -1,21 +1,26 @@
...
-  }
-
-  return { isValid: true };
-}
-
 export function useNonZeroBn(initialValue: BN | number = BN_ZERO): ValidFormField<BN> {
+  const { t } = useTranslation();
   const value = useMemo(() => bnToBn(initialValue), [initialValue]);
 
+  const isValid = useCallback(
```
[the remark this answers is not in the document]
Comment (kwingram25) [reply src/ui/hooks/useNonZeroBn.ts:14]: Moved because it uses the t function now

File: src/ui/hooks/useLocalStorage.ts (hunk 1 above)
Comment (kwingram25) [reply src/ui/hooks/useLocalStorage.ts:11]: Strings like 'en' were being stored with double quotes because of JSON.stringify. Fixed

File: src/ui/components/Transactions.tsx [hunk 3]
```diff
@@ -4,60 +4,53 @@
 import { BellIcon, XIcon } from '@heroicons/react/outline';
 import React from 'react';
 import { NotificationIcon } from './common/NotificationIcon';
-import type { TransactionsState } from 'types';
+import type { QueuedTxOptions, TransactionsState } from 'types';
```
[the remark this answers is not in the document]
Comment (kwingram25) [reply src/ui/components/Transactions.tsx:7]: Using the spread operator on React nodes was causing errors with i18next-scanner. In any case this seems a more logical way to implement it.

**Comment by kwingram25** [comment]
Remember to use the <Trans> component when there are interpolated components instead of translating each piece.

Cleaned up any conflicts but there are some strings still untranslated. I think I everything has been moved from the markdown to en.json, however

**Comment by statictype** [comment]
i don't like doing refusing people's work, this is why i agreed to this PR even if it was never discussed to translate the app in the first place.
But I'm just thinking of how much it will complicate our lives in the future for no good reason.
Every piece of text needs to have at least a german translation now. I'm pretty sure our users who don't know english very well, already have some kind of strategy to translate web content if they're interested in smart contracts.
Keeping everything in english, including tooltips and other hardcoded text should be absolutely fine for our use case.
I'm sorry for agreeing to this and strongly advise not to merge this PR.  @achimcc @kwingram25
